eZ/Publish/Core/FieldType/BinaryBase/Type.php

```diff
@@ -347,20 +347,7 @@ public function validateValidatorConfiguration( $validatorConfiguration )
             switch ( $validatorIdentifier )
             {
                 case 'FileSizeValidator':
-                    if ( !isset( $parameters['maxFileSize'] ) )
-                    {
-                        $validationErrors[] = new ValidationError(
-                            "Validator %validator% expects parameter %parameter% to be set.",
-                            null,
-                            array(
-                                "validator" => $validatorIdentifier,
-                                "parameter" => 'maxFileSize',
-                            ),
-                            "[$validatorIdentifier][maxFileSize]"
-                        );
-                        break;
-                    }
-                    if ( !is_int( $parameters['maxFileSize'] ) && $parameters['maxFileSize'] !== false )
+                    if ( array_key_exists( "maxFileSize", $parameters ) && !is_int( $parameters['maxFileSize'] ) && $parameters['maxFileSize'] !== null )
```

Comment:
As I said, I've tried this already, and !array_key_exists( 'maxFileSize', $parameters ) always triggers when the type defaults to null. Something else is broken, possibly including my sanity.